contracts/XINV.sol

```diff
@@ -0,0 +1,965 @@
+pragma solidity ^0.5.16;
+
+import "./ComptrollerInterface.sol";
+import "./CTokenInterfaces.sol";
+import "./ErrorReporter.sol";
+import "./Exponential.sol";
+import "./EIP20Interface.sol";
+import "./EIP20NonStandardInterface.sol";
+import "./SafeMath.sol";
+
+/**
+ * @title Compound's CToken Contract
+ * @notice Abstract base for CTokens
+ * @author Compound
+ */
+contract xInvCore is CTokenInterface, Exponential, TokenErrorReporter {
+    /**
+     * @notice Initialize the money market
+     * @param comptroller_ The address of the Comptroller
+     * @param initialExchangeRateMantissa_ The initial exchange rate, scaled by 1e18
+     * @param name_ EIP-20 name of this token
+     * @param symbol_ EIP-20 symbol of this token
+     * @param decimals_ EIP-20 decimal precision of this token
+     */
+    function initialize(ComptrollerInterface comptroller_,
+                        uint initialExchangeRateMantissa_,
+                        string memory name_,
+                        string memory symbol_,
+                        uint8 decimals_) public {
+        require(msg.sender == admin, "only admin may initialize the market");
```

Comment:
Admin can initialize as many times as they want? Can we rename this? Initialize is usually reserved for proxies.

Reply:
The original CToken can be used as an immutable or upgradable contract.
I changed `initialize` visibility to internal to prevent re-initialization by admin